packages/context/src/context.ts

```diff
@@ -281,3 +344,57 @@ export class Context {
     return json;
   }
 }
+
+/**
+ * Apply a mapping function to the given iterator to produce a new iterator
+ * with mapped values as its entries
+ *
+ * @param iterator An iterable object
+ * @param mapper A function maps an entry to a new value
+ */
+function* map<T, V>(
```

Comment:
Please extract `map`, `takeUntil` and `reduce` into a new file, e.g. `src/async-iterables.ts`.

AFAICT, there are no (unit) tests for these three methods, please add some.

Reply:
Oh, good catch. They are left over from the cherry-pick and should be removed.